engine/class_modules/priest/sc_priest_shadow.cpp

```diff
@@ -1408,13 +1406,12 @@ struct void_torrent_t final : public priest_spell_t
     energize_type     = action_energize::PER_TICK;
     energize_resource = RESOURCE_INSANITY;
     energize_amount   = insanity_gain;
-
-    child_dp->background = true;
   }
 
+  // DoT duration is fixed at 3s
   timespan_t composite_dot_duration( const action_state_t* ) const override
   {
-    return timespan_t::from_seconds( 4.0 );
+    return data().duration();
```

Comment:
This should just be `return dot_duration;`
Is there a reason why we override tick_time? The default implementation of tick_time appears to be identical except it gets the haste scaler from the action state instead of `priest().composite_spell_haste()`

Reply:
good call, probably just leftover code